commands/show_test.go

```diff
@@ -0,0 +1,108 @@
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+func TestBlockDaemon(t *testing.T) {
+	tf.IntegrationTest(t)
+
+	t.Run("show block <cid-of-genesis-block> returns human readable output for the filecoin block", func(t *testing.T) {
+		d := makeTestDaemonWithMinerAndStart(t)
+		defer d.ShutdownSuccess()
+
+		// mine a block and get its CID
+		minedBlockCidStr := th.RunSuccessFirstLine(d, "mining", "once")
+
+		// get the mined block by its CID
+		output := d.RunSuccess("show", "block", minedBlockCidStr).ReadStdoutTrimNewlines()
+
+		assert.Contains(t, output, "Block Details")
+		assert.Contains(t, output, "Weight: 0")
+		assert.Contains(t, output, "Height: 1")
+		assert.Contains(t, output, "Nonce:  0")
+	})
+
+	t.Run("show block <cid-of-genesis-block> --enc json returns JSON for a filecoin block", func(t *testing.T) {
+		d := th.NewDaemon(t,
+			th.KeyFile(fixtures.KeyFilePaths()[0]),
+			th.WithMiner(fixtures.TestMiners[0])).Start()
+		defer d.ShutdownSuccess()
+
+		// mine a block and get its CID
+		minedBlockCidStr := th.RunSuccessFirstLine(d, "mining", "once")
+
+		// get the mined block by its CID
+		blockGetLine := th.RunSuccessFirstLine(d, "show", "block", minedBlockCidStr, "--enc", "json")
+		var blockGetBlock types.Block
+		require.NoError(t, json.Unmarshal([]byte(blockGetLine), &blockGetBlock))
+
+		// ensure that we were returned the correct block
+
+		require.Equal(t, minedBlockCidStr, blockGetBlock.Cid().String())
+
+		// ensure that the JSON we received from block get conforms to schema
+
+		requireSchemaConformance(t, []byte(blockGetLine), "filecoin_block")
+	})
+}
+
+func TestShowDeal(t *testing.T) {
+	tf.IntegrationTest(t)
+
+	minerDaemon := th.NewDaemon(t,
+		th.WithMiner(fixtures.TestMiners[0]),
+		th.KeyFile(fixtures.KeyFilePaths()[0]),
+		th.DefaultAddress(fixtures.TestAddresses[0]),
+		th.AutoSealInterval("1"),
+	).Start()
+	defer minerDaemon.ShutdownSuccess()
+
+	clientDaemon := th.NewDaemon(t,
+		th.KeyFile(fixtures.KeyFilePaths()[1]),
+		th.DefaultAddress(fixtures.TestAddresses[1]),
+	).Start()
+	defer clientDaemon.ShutdownSuccess()
+
+	minerDaemon.RunSuccess("mining", "start")
+	minerDaemon.UpdatePeerID()
+
+	minerDaemon.ConnectSuccess(clientDaemon)
+
+	addAskCid := minerDaemon.MinerSetPrice(fixtures.TestMiners[0], fixtures.TestAddresses[0], "20", "10")
+
+	t.Run("when deal exists shows deal info", func(t *testing.T) {
+		clientDaemon.WaitForMessageRequireSuccess(addAskCid)
+		dataCid := clientDaemon.RunWithStdin(strings.NewReader("HODLHODLHODL"), "client", "import").ReadStdoutTrimNewlines()
+
+		proposeDealOutput := clientDaemon.RunSuccess("client", "propose-storage-deal", fixtures.TestMiners[0], dataCid, "0", "5").ReadStdoutTrimNewlines()
+
+		splitOnSpace := strings.Split(proposeDealOutput, " ")
+
+		dealCid := splitOnSpace[len(splitOnSpace)-1]
+
+		showdeal := minerDaemon.RunSuccess("show", "deal", dealCid).ReadStdoutTrimNewlines()
```

Comment:
(Disclaimer this comment doesn't imply any more work to this PR)
It's unfortunate it takes us setting up a small network to get a deal datastructure.  I think a long term goal is to have tooling exposing data structure construction, signing and adding to the node.  In that better world this test setup would consist of creating two keys / wallets, deal construction would happen with a few commands and loading the deal to the node.